### PR TITLE
AP_TemperatureSensor: Remove unused headers and double quotes

### PR DESCRIPTION
**Pull request by muramura**
I have found a double quote in the header file and the CPP file.
I found some unused headers.
I sorted out unused headers and double quotes.

### CODE REVIEW

File: libraries/AP_TemperatureSensor/TSYS01.h [hunk 1]
```diff
@@ -5,8 +5,6 @@
 #pragma once
 
 #include <AP_HAL/AP_HAL.h>
-#include <AP_HAL/Semaphores.h>
-#include <AP_HAL/Device.h>
```
Comment (khancyr) [inline libraries/AP_TemperatureSensor/TSYS01.h]: this one is need for line 25

File: libraries/AP_TemperatureSensor/TSYS01.h [hunk 2]
```diff
@@ -5,8 +5,6 @@
 #pragma once
 
 #include <AP_HAL/AP_HAL.h>
-#include <AP_HAL/Semaphores.h>
```
Comment (khancyr) [inline libraries/AP_TemperatureSensor/TSYS01.h]: this one should be in .cpp

File: libraries/AP_TemperatureSensor/TSYS01.cpp [hunk 3]
```diff
@@ -1,8 +1,8 @@
 #include "TSYS01.h"
 
-#include <utility>
 #include <stdio.h>
-#include <AP_HAL/AP_HAL.h>
+#include <utility>
+#include <AP_HAL/Semaphores.h>
```
Comment (peterbarker) [inline libraries/AP_TemperatureSensor/TSYS01.cpp:5]: Why is this required in the cpp file?

Comment (khancyr) [reply libraries/AP_TemperatureSensor/TSYS01.cpp:5]: yep, in fact it is not even needed ...my bad

File: libraries/AP_TemperatureSensor/TSYS01.cpp [hunk 4]
```diff
@@ -1,8 +1,7 @@
 #include "TSYS01.h"
 
-#include <utility>
 #include <stdio.h>
-#include <AP_HAL/AP_HAL.h>
```
Comment (peterbarker) [inline libraries/AP_TemperatureSensor/TSYS01.cpp:5]: Uh.  We're creating an `AP_HAL::HAL` several lines down.  We really ought to include its header.

File: libraries/AP_TemperatureSensor/TSYS01.h [hunk 5]
```diff
@@ -5,7 +5,6 @@
...
```
Comment (peterbarker) [inline libraries/AP_TemperatureSensor/TSYS01.h:8]: The .cpp file is making use of semaphores; we really ought to include the header in the .cpp file.

**Comment by khancyr** [comment]
Finally, this doesn't bring any benefit... closing
